internal/search/job/jobutil/job.go

```diff
@@ -762,7 +762,7 @@ func optimizeJobs(baseJob job.Job, inputs *run.SearchInputs, q query.Basic) (job
 				}
 				return currentJob
 
-			case *commit.CommitSearch:
+			case *commit.CommitSearchJob:
 				if exists("Commit") || exists("Diff") {
```

Comment:
We'll need to update any references to the name string as well

```suggestion
				if exists("CommitSearchJob") || exists("DiffSearchJob") {
```

(sorry for the using strings as identification here. the stuff we worked on today should  make this unnecessary)

Reply:
Ah, thanks. Fixed